clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1313,8 +1309,6 @@ class KernelObjVisitor {
                   QualType FieldTy, HandlerTys &... Handlers) {
     if (Util::isSyclSpecialType(FieldTy))
       KF_FOR_EACH(handleSyclSpecialType, Field, FieldTy);
-    else if (Util::isSyclHalfType(FieldTy))
```

Comment:
I'm actually convinced we need the handlers - if a struct containing `half ` is decomposed (because of other special types) or `half` is a root level arguments, it will not be handled now.